main.js

```diff
@@ -0,0 +1,191 @@
+// selet elements in DOM
+
+const form = document.querySelector("#itemForm");
+const itemInput = document.querySelector("#itemInput");
+const itemList = document.querySelector("#itemList");
+const messageDiv = document.querySelector("#message");
+const clearButton = document.querySelector("#clearBtn");
+const filters = document.querySelectorAll(".nav-item");
+
+// create an empty array for tasks(items).
+let todoItems = [];
+
+// showing alert when we add item, or errors.
+const showAlert = function (message, msgClass) {
+  //msgClass = type of message, ie. success, danger, error, etc.
+  console.log("msg");
+  messageDiv.innerHTML = message;
+  messageDiv.classList.add(msgClass, "show");
+  messageDiv.classList.remove("hide");
+  setTimeout(() => {
+    messageDiv.classList.remove("show", msgClass);
+    messageDiv.classList.add("hide");
+  }, 2000);
+  return;
+};
+
+// filter tab tasks
+const getFilteredItems = function (type) {
+  let filterItems = [];
+  switch (type) {
+    case "remaining":
+      filterItems = todoItems.filter((item) => !item.isDone);
+      break;
+    case "finished":
+      filterItems = todoItems.filter((item) => item.isDone);
+      break;
+    default:
+      filterItems = todoItems;
+  }
+  setList(filterItems);
+};
+
+//delete item
+const removeItem = function (item) {
+  const removeIndex = todoItems.indexOf(item);
+  todoItems.splice(removeIndex, 1);
+};
+
+// handle item
+const handleItem = function (itemData) {
+  //getting all tasks in HTML code.
+  //itemData is from JS array. It is an object of text, isdone, addedAt.
+  const items = document.querySelectorAll(".list-group-item");
+
+  items.forEach((item) => {
+    if (
+      item.querySelector(".task").getAttribute("data-time") == itemData.addedAt
+    ) {
+      // done
+      item.querySelector("[data-done]").addEventListener("click", function (e) {
+        e.preventDefault();
+        const itemIndex = todoItems.indexOf(itemData);
+        const currentItem = todoItems[itemIndex];
+        const currentClass = currentItem.isDone
+          ? "bi-check-circle-fill"
+          : "bi-check-circle";
+
+        //changing status of this current task.
+        currentItem.isDone = currentItem.isDone ? false : true;
```

Comment:
if your intention is to toggle the value, you can use ! (not) operator instead of ternary
`x = !x`